### PR TITLE
Clarify Gateway Listener Distinct definitions

### PR DESCRIPTION
**Pull request by youngnick**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

This PR updates the Listener godocs to clarify:

- what "distinct" means for each currently supported Protocol
- what ports are Core support for HTTPRoute and TLSRoute (this referred to the outdated concept of conformance profiles based on the supported protocols).

Updates #3297 

**Does this PR introduce a user-facing change?**:
```release-note
Clarified what Gateway Listeners being Distinct means
```

### CODE REVIEW

File: apis/v1/gateway_types.go
Comment (robscott) [inline apis/v1/gateway_types.go:123]: I think TLS may not really be a differentiator here

Comment (youngnick) [reply apis/v1/gateway_types.go:123]: Dang, you're right.

I'll add a point about how TLS may affect _validity_ since it doesn't affect _distinctiveness_ though.

File: apis/v1/gateway_types.go
Comment (youngnick) [reply apis/v1/gateway_types.go:190]: I added some more words to clarify this, thanks!

File: apis/v1/gateway_types.go
Comment (youngnick) [reply apis/v1/gateway_types.go:126]: I wanted to call out TLS in particular because I think it's more likely to be misunderstood.

Good call about calling out the actual field name, I've added that.

File: apis/v1/gateway_types.go
Comment (youngnick) [reply apis/v1/gateway_types.go:169]: Agreed.

**Review by mlavacca — APPROVED** [review]
/approve

File: apis/v1/gateway_types.go
Comment (youngnick) [inline apis/v1/gateway_types.go:127]: @robscott _Very_ important update here to close out #3297, what happens when a port is shared between HTTPRoute and TCPRoute.

This _is_ a behavior change, in that the behavior was previously unspecified, although I'm pretty confident that no implementations would be able to do anything different.

Comment (youngnick) [reply apis/v1/gateway_types.go:127]: Of course, that makes sense, thanks, will update.

File: apis/v1/gateway_types.go
Comment (bjee19) [inline apis/v1/gateway_types.go:202]: nit:
```suggestion
	// Implementations that _do_ support Listener Isolation SHOULD claim support
```

**Comment by robscott** [comment]
Soak time has passed, will LGTM once the comment above is resolved.

/hold cancel

**Pull request merged**
